src/test/scala/edu/ie3/simona/agent/grid/ThermalGridIT.scala

```diff
@@ -0,0 +1,857 @@
+              qDot should equalWithTolerance(0.011.asMegaWatt)
+              energy should equalWithTolerance(0.asMegaWattHour)
+            case _ =>
+              fail(
+                "Expected a ThermalHouseResult and a ThermalStorageResult but got something else"
+              )
+          }
+        }
+
+      scheduler.expectMessage(Completion(heatPumpAgent, Some(0L)))
+
+      /* TICK 3417
+      Storage is fully heated up
+      House demand heating : requiredDemand = 0.0 kWh, additionalDemand = 17.37 kWh
+      House demand water   : tba
+      ThermalStorage       : requiredDemand = 0.0 kWh, additionalDemand = 0.0 kWh
+      DomesticWaterStorage : tba
+      Heat pump: stays on since it was on and the house as additional demand
+       */
+
+      heatPumpAgent ! Activation(3417)
+
+      resultListener.expectMessageType[ParticipantResultEvent] match {
+        case ParticipantResultEvent(hpResult) =>
+          hpResult.getInputModel shouldBe typicalHpInputModel.getUuid
+          hpResult.getTime shouldBe 3417.toDateTime
+          hpResult.getP should equalWithTolerance(pRunningHp)
+          hpResult.getQ should equalWithTolerance(
+            qRunningHp
+          )
+      }
+
+      Range(0, 2)
+        .map { _ =>
+          resultListener.expectMessageType[ResultEvent]
+        }
+        .foreach { case ResultEvent.ThermalResultEvent(thermalUnitResult) =>
+          thermalUnitResult match {
+            case ThermalHouseResult(
+                  time,
+                  inputModel,
+                  qDot,
+                  indoorTemperature,
+                ) =>
+              inputModel shouldBe typicalThermalHouse.getUuid
+              time shouldBe 3417.toDateTime
+              qDot should equalWithTolerance(0.011.asMegaWatt)
+              indoorTemperature should equalWithTolerance(
+                19.6835196903292.asDegreeCelsius
+              )
+
+            case CylindricalThermalStorageResult(
+                  time,
+                  inputModel,
+                  qDot,
+                  energy,
+                ) =>
+              inputModel shouldBe typicalThermalStorage.getUuid
+              time shouldBe 3417.toDateTime
+              qDot should equalWithTolerance(0.asMegaWatt)
+              energy should equalWithTolerance(0.01044.asMegaWattHour)
+            case _ =>
+              fail(
+                "Expected a ThermalHouseResult and a ThermalStorageResult but got something else"
+              )
+          }
+        }
+
+      // FIXME? Why next tick 3417?
+      scheduler.expectMessage(Completion(heatPumpAgent, Some(3417)))
+
+      /* TICK 7200
+      New weather data (unchanged) incoming
+      House demand heating : requiredDemand = 0.0 kWh, additionalDemand = 8.41 kWh
+      House demand water   : tba
+      ThermalStorage       : requiredDemand = 0.0 kWh, additionalDemand = 0.0 kWh
+      DomesticWaterStorage : tba
+      Heat pump: stays on
```

Comment:
... I guess for the same reason as last activation?

Reply:
Yes, in detail we just continue as before. We got triggered because there was new weather data.

I added this explanation to the comments.